sdk/storage/AzureStorageBlob/Source/Data/BlobTransfer+CoreDataClass.swift

```diff
@@ -83,6 +83,58 @@ public class BlobTransfer: NSManagedObject, TransferImpl {
         }
     }
 
+    // FIXME: The fact that BlobProperties is a struct causes serious problems here. It will always return nil if
+    // allowed to be optional.
+    internal var properties: BlobProperties {
```

Comment:
This can be revisited later, if necessary. Since `BlobProperties` is a struct, making this optional always returns a nil value. This was the quickest solution. Another solution (not convinced it is better) would be to convert this to a class.

Reply:
I don't quite follow what's going on here. UploadBlobOptions is also a struct, doesn't `uploadOption` exhibit the same behavior?

Reply:
I don't understand why this would be the case, or why it would matter between a struct and a class. Is the issue related to some semantics of how/when the getter for a computed property that returns a struct is called? Some sort of weird optimization for structs?

Reply:
I don't rightly understand why this would be the case either, but it was definitely the culprit. Building the struct and examining its value right before returning would show the correct values and then stepping to the next line, immediately after the return, would inexplicably be nil. Making it non-optional worked (no idea why), and making it a class (as I did with the Options) worked. It may be related to the fact that structs are value types instead of reference types.

Reply:
Well.. to implement the memoization of properties and upload/download options, the most straightforward solution was to make the options classes into structs anyways and then make them non-optional. Doing the memoization required the objects to conform to Equatable. For structs, that can be automatically synthesized, but not for classes. Again, I'll reconsider all of this in a future PR, but for now, the side effect is that if you called `.downloadOptions` on an upload transfer, instead of getting `nil` you'll get a new `DownloadBlobOptions` struct. Certainly non-ideal.

Reply:
Sounds good, I think we should definitely get to the bottom of what's going on here, but we can address that in a future PR.